### PR TITLE
Add open_browser and server_host config options

### PR DESCRIPTION
**Pull request by Dobby233Liu**
## Related Issue/Addition to code

Per #491

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Proposed Changes

- Read config at the very start
- Add `open_browser` config option
- Add `server_host` config option
- Add `x_action` parameter to utils.choice

Options are explained in the new default config

## Additional Info

See #491

## Checklist:

- [x] My code follows the style guidelines of this project and have read CONTRIBUTING.md
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
	I edited the configuration only
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: Scripts/auth.py [hunk 1]
```diff
@@ -69,9 +72,37 @@ def get_authenticated_service():
...
+          sys.exit()
+      host = config["server_host"]
+      if host == "auto":
+        # See https://stackoverflow.com/a/166589 , this seems to be a safe option to get the IP address;
+        # This program needs internet anyway
+        try:
+          test_socket = socket(AF_INET, SOCK_DGRAM)
+          test_socket.connect(("8.8.8.8", 80))
+          host = test_socket.getsockname()[0]
+          print(f"Using host: {F.LIGHTGREEN_EX}{host}{S.R}")
```
Comment (Dobby233Liu) [inline Scripts/auth.py:90]: These are kinda excessive prints, should I remove them?

**Comment by ThioJoe** [comment]
are you sure just removing the "authorization_prompt_message" parameter from the line below doesn't achieve what you're looking for?
`creds = flow.run_local_server(port=0, authorization_prompt_message="Waiting for authorization. See message above.")`

**Comment by Dobby233Liu** [comment]
> Are you sure just removing the "authorization_prompt_message" parameter from the line below doesn't achieve what you're looking for? `creds = flow.run_local_server(port=0, authorization_prompt_message="Waiting for authorization. See message above.")`
> 
> Pretty sure that makes it show the url

I want to keep the "Waiting for authorization. See message above." message

**Comment by ThioJoe** [comment]
> > Are you sure just removing the "authorization_prompt_message" parameter from the line below doesn't achieve what you're looking for? `creds = flow.run_local_server(port=0, authorization_prompt_message="Waiting for authorization. See message above.")`
> > Pretty sure that makes it show the url
> 
> I want to keep the "Waiting for authorization. See message above." message

Hm I see... Well you could just have a print statement that prints right before the url. Unless you mean you want to catch the URL into a variable to use for another purpose elsewhere, and not necessarily have it just printed to the console?

**Comment by Dobby233Liu** [comment]
No, and I intentionally made it like
```
Please open this URL in a browser to login:
[url]


Waiting for authorization. See message above.
```
In the code, so the message prints under the url like in the original. Saving the url makes no sense

**Comment by ThioJoe** [comment]
I see, well since removing the 'message' parameter solves your problem, I don't think it would make sense to add all the code from the PR just for an aesthetic change.

**Pull request closed without merging**